java/poi/src/main/java/com/example/App.java

```diff
@@ -0,0 +1,82 @@
+import java.sql.Connection;
+import java.sql.Statement;
+import java.sql.ResultSet;
+import java.sql.ResultSetMetaData;
+import java.sql.Types;
+
+import com.ibm.as400.access.AS400JDBCDriver;
+import com.ibm.as400.access.AS400;
+
+import java.util.Properties;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+
+import java.lang.Exception;
+
+import org.apache.poi.ss.usermodel.WorkbookFactory;
+import org.apache.poi.ss.usermodel.Workbook;
+import org.apache.poi.ss.usermodel.Sheet;
+import org.apache.poi.ss.usermodel.Row;
+import org.apache.poi.ss.usermodel.Cell;
+
+
+public class App
+{
+    public static void main( String[] args ) throws Exception
+    {
+      Properties props = new Properties();
+      props.load(new FileInputStream(".env"));
+
+      String HOST = props.getProperty("DB_HOST");
+      String USER = props.getProperty("DB_USER");
+      String PASS = props.getProperty("DB_PASS");
+      String query = "SELECT CUSNUM, LSTNAM, BALDUE, CDTLMT FROM QIWS.QCUSTCDT";
+
+      AS400JDBCDriver driver = new AS400JDBCDriver();
+      AS400 as400 = new AS400(HOST, USER, PASS);
+
+      Connection connection = driver.connect(as400);
+      Statement statement = connection.createStatement();
+      ResultSet results = statement.executeQuery(query);
+      ResultSetMetaData metadata = results.getMetaData();
+
+      // create workbook and workshee
+      Workbook workbook = WorkbookFactory.create(true);
+      Sheet sheet = workbook.createSheet("Customers");
+
+      // add the header
+      Row header = sheet.createRow(0);
+      int columnCount = metadata.getColumnCount();
+
+      for (int i = 0; i < columnCount; i++) {
+        // column name start with index 1
+        header.createCell(i).setCellValue(metadata.getColumnName(i + 1));
+      }
+
+      // fill in the worksheet
+      for (int r = 1; results.next(); r++) {
+        Row row = sheet.createRow(r);
+        for (int c = 0; c < columnCount; c++) {
+          Cell cell = row.createCell(c);
+          // column name and type start with index 1
+          String columnName = metadata.getColumnName(c + 1);
+          int columnType = metadata.getColumnType(c + 1);
+
+          switch (columnType) {
+            case Types.NUMERIC:
+              row.createCell(c).setCellValue(results.getDouble(columnName));
+            break;
+            case Types.CHAR:
```

Comment:
Should we make this a `default:` block?

Reply:
OK, should it be dummy default block that does not do anything?

Reply:
In our example the only 2 types are Numeric and Character data. If default case was added not sure it would be useful.

Reply:
As currently written, other types won't get written to the spreadsheet at all. If the default case were to write a string representation (same as we do for `Types.CHAR`, that would be a good fallback. so the `Types.CHAR` case can double as the default. 
Even cleaner would be to not use case at all. Just:
```java
 if(Types.NUMERIC == columnType) { 
    row.createCell(c).setCellValue(results.getDouble(columnName));
} else {
    row.createCell(c).setCellValue(results.getString(columnName));
}

Reply:
Yeah avoiding the switch would be cleaner in this case. If someone had many different column types the switch statement would be easier then a bunch of `if conditions`. I would like to keep the switch statement here but add a comment to add additional cases as needed if you have additional types.